client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs

```diff
@@ -30,33 +33,64 @@ public static class AutoRegistrationInMonitoring
         {
             if (configuration == null)
                 throw new ArgumentNullException(nameof(configuration));
+
+            string disableAutoRegistrationStr = configuration[_disableAutoRegistrationEnvVarName];
+            if (bool.TryParse(disableAutoRegistrationStr, out bool disableAutoRegistration) && disableAutoRegistration)
+                return;
+
             if (string.IsNullOrWhiteSpace(monitoringServiceUrl))
                 throw new ArgumentException("Argument is empty", nameof(monitoringServiceUrl));
+
             if (log == null)
```

Comment:
better to do preconditions check at the start of the method

Reply:
This is the start of the method. And all the code before this does checks before actual functionality